src/MQTTAsyncUtils.c

```diff
@@ -2058,23 +2084,31 @@ thread_return_type WINAPI MQTTAsync_receiveThread(void* n)
 		pack = MQTTAsync_cycle(&sock, timeout, &rc);
 		MQTTAsync_lock_mutex(mqttasync_mutex);
 		if (MQTTAsync_tostop)
+		{
+			MQTTAsync_freePack(pack);
 			break;
+		}
 		timeout = 1000L;
 
 		if (sock == 0)
+		{
+			MQTTAsync_freePack(pack);
```

Comment:
_pack_ is going out of scope without transfer. (pack may or may not point to memory)